cpp/include/kvikio/posix_io.hpp

```diff
@@ -208,6 +208,9 @@ inline std::size_t posix_io(int fd,
 /**
  * @brief Read main memory from disk using POSIX
  *
+ * If `size` or `file_offset` isn't aligned with `page_size` then
+ * `fd` cannot have been opened with the `O_DIRECT` flag.
```

Comment:
Does this mean we should be doing some kind of error handling when opening with `O_DIRECT`?

Reply:
I don't think so, `posix_io` is internal so will move in to `namespace kvikio::detail` as well

Reply:
Sorry I mean when `FileHandle` is created, is there a risk an exception is thrown because `O_DIRECT` doesn't work? If so, should we be handling this somehow there?

Reply:
> Sorry I mean when `FileHandle` is created, is there a risk an exception is thrown because `O_DIRECT` doesn't work? If so, should we be handling this somehow there?

No this cannot happen. My plan is that `posix_io` will be given two file descriptors, one with and one with the `O_DIRECT` flag. It will then use the `O_DIRECT` file for aligned IO only.